File: app/src/main/java/com/bailout/stickk/ubi4/ble/SampleGattAttributes.java
package com.bailout.stickk.ubi4.ble;


import java.util.HashMap;
import java.util.UUID;

/**
 * This class includes a small subset of standard GATT attributes for demonstration purposes.
 */
public class SampleGattAttributes {
    private static final HashMap<String, String> attributes = new HashMap();
    public static String MAIN_CHANNEL = "43680201-4d74-1001-726b-526f64696f6e";
    public static String A0_MAIN_CHANNEL = "0000fff4-0000-1000-8000-00805f9b34fb";
    public static UUID A0_MAIN_CHANNEL_UUID = UUID.fromString(A0_MAIN_CHANNEL);
    public static String A0_MAIN_SERVICE = "0000fff0-0000-1000-8000-00805f9b34fb";
    public static UUID A0_MAIN_SERVICE_UUID = UUID.fromString(A0_MAIN_SERVICE);

    // Sample Commands.
    public static Boolean SHOW_EVERYONE_RECEIVE_BYTE = false;
    public static String READ = "READ";
    public static String WRITE = "WRITE";
    public static String NOTIFY = "NOTIFY";

    public static String lookup(String uuid, String defaultName) {
        String name = attributes.get(uuid);
        return name == null ? defaultName : name;
    }
}

File: shared/src/commonMain/kotlin/com/bailout/stickk/ubi4/ble/SampleGattAttributes.kt
package com.bailout.stickk.ubi4.ble

object SampleGattAttributes {

    private val attributes: MutableMap<String, String> = mutableMapOf()

//    const val MAIN_CHANNEL_SERVICE = "43686172-4D74-1001-726B-526F64696F6E"
//    const val MAIN_CHANNEL_CHARACTERISTIC = "43680201-4D74-1001-726B-526F64696F6E"

    const val MAIN_CHANNEL_SERVICE =        "43686172-4d74-1001-726b-526f64696f6e"
    const val MAIN_CHANNEL_CHARACTERISTIC = "43680201-4d74-1001-726b-526f64696f6e"

    const val SENSORS_STREAM_UUID =         "43680201-4d74-1001-726b-526f64696f6e"
    const val SERIALPORTCHAR_UUID =         "43681000-4d74-1001-726b-526f64696f6e"

    var showEveryoneReceiveByte: Boolean = false

    const val READ = "READ"
    const val WRITE = "WRITE"
    const val NOTIFY = "NOTIFY"

    fun lookup(uuid: String, defaultName: String): String =
        attributes[uuid] ?: defaultName

    /**
     * Позволяет добавить или обновить UUID → имя для lookup.
     */
    fun register(uuid: String, name: String) {
        attributes[uuid] = name
    }
}